src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -280,19 +283,66 @@ private static boolean isDate(String token) {
   }
 
   /**
-   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
-   * it exists, is added to the builder.
+   * Searches the raw text in the builder for the total price using multiple heuristics. If any find
+   * a price, it is added to the builder.
    */
   private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
-    // Assume that the largest price on the receipt is the total price.
+    if (!findPriceAfterTotal(analysisBuilder)) {
+      findLargestPrice(analysisBuilder);
+    }
+  }
+
+  /**
+   * Checks the raw text in the builder for a price appearing after the word "total" and adds it to
+   * the builder if found.
+   *
+   * @return whether or not a price was added.
+   */
+  private static boolean findPriceAfterTotal(AnalysisResults.Builder analysisBuilder) {
+    ImmutableList<String> relevantTokens =
+        getTokensFromRawText(analysisBuilder.getRawText().get())
+            .filter(token -> isPrice(token) || containsTotal(token))
+            .collect(ImmutableList.toImmutableList());
+
+    double price = 0;
+    boolean priceFound = false;
+    boolean priceFoundAfterMostRecentTotal = true;
```

Comment:
If you can, it's better to not maintain state over a loop to simplify the code. In this case, I recommend finding all the indexes of the "total" text. Then, get the price after each index.

Reply:
Alternatively, could we use regex to help find all "total" and price pairs and then we could just grab the last pair found?

Reply:
Hey Ava, I know in our 1:1 we decided it was okay to leave this alone since we were low on time. After discussing with Tony, we decided we would like you to fix this. It would look better for the hiring committee.